pkg/app/piped/platformprovider/kubernetes/kubectl.go

```diff
@@ -223,3 +225,22 @@ func (c *Kubectl) Get(ctx context.Context, kubeconfig, namespace string, r Resou
 	}
 	return ms[0], nil
 }
+
+func (c *Kubectl) CreateNamespace(ctx context.Context, kubeconfig, namespace string) (err error) {
+	args := make([]string, 0, 7)
+	if kubeconfig != "" {
+		args = append(args, "--kubeconfig", kubeconfig)
+	}
+	args = append(args, "create", "namespace", namespace)
+
+	cmd := exec.CommandContext(ctx, c.execPath, args...)
+	out, err := cmd.CombinedOutput()
+
+	if strings.Contains(string(out), errAlreadyExistsLiteral) {
+		return errReplaceAlreadyExists
```

Comment:
Rename due to https://github.com/pipe-cd/pipecd/pull/4504#discussion_r1271260320